src/services/item/plugins/etherpad/README.md

```diff
@@ -0,0 +1,243 @@
+# graasp-plugin-etherpad
+
+This Fastify plugin implements the glue between the Etherpad external service and the Graasp platform. It exposes endpoints to manage Etherpad items in the Graasp object model, and provides authorization mapping between the permissions in Graasp to those in Etherpad.
+
+The plugin communicates with the Etherpad server through the [HTTP API](https://etherpad.org/doc/v1.8.18/#index_http-api). You may find the [Etherpad documentation](https://etherpad.org/doc/v1.8.18/) useful as well.
+
+We use our fork of the `etherpad-api` library at https://github.com/graasp/etherpad-api to facilitate the usage of the HTTP API in Typescript.
+
+## Prerequisites
+
+An instance of the Etherpad (lite) server should be available on the network to this plugin. The list of options to connect to it is available [here](types.ts).
+
+#### A note on the Etherpad server API key
+
+A symmetric key is required between the Etherpad server and the Graasp back-end. We describe below how to add it to the Etherpad server instance, and the key should be copied to this plugin's input options, e.g. as secret environment variable.
+
+Note that the master API key must be placed in the root folder of the Etherpad server deployment, see the related [documentation](https://etherpad.org/doc/v1.8.18/#index_authentication):
+
+> Authentication works via a token that is sent with each request as a post parameter. There is a single token per Etherpad deployment. This token will be random string, generated by Etherpad at the first start. It will be saved in APIKEY.txt in the root folder of Etherpad. Only Etherpad and the requesting application knows this key. Token management will not be exposed through this API.
+
+The key format must be a hex string of 64 characters (i.e. regex `[a-f\d]{64}`).
+
+In the Graasp devcontainer, the key is bind-mounted from [this file](../../../../../.devcontainer/etherpad/devApiKey.txt) and the container is already available over the virtual network from the [docker-compose definition](../../../../../.devcontainer/docker-compose.yml).
+
+In a production container environment, you should either bind-mount a secret key, or build your own etherpad image using the `etherpad/etherpad` image as base and write your own secret file, see the [Dockerfile](Dockerfile) as an example.
```

Comment:
```
you should either bind-mount a secret key
```
I don't get how this could be equivalent to building an image. I guess you mean a remote service already exists?

Reply:
I've tried clarifying it a bit. So in the Etherpad server installation, the key should at exist the root path and named APIKEY.txt. In a containerized deployment, the Etherpad image should be instantiated with this file at /opt/etherpad/APIKEY.txt, in which case you have 2 options: either you bind-mount this key from the host filesystem to the path, or you build your copy of the etherpad container image with the key already baked in.

In our case, since we do not use bind mounts on the ECS instance of the etherpad container, we chose to build our own image with the key using the Dockerfile. Then the container with the key is shipped to ECR and ECS just picks up the image and spins up a container with it.